wandb/sdk/wandb_init.py

```diff
@@ -171,47 +178,55 @@ def setup(self, kwargs) -> None:
             magic_install(kwargs)
 
         # handle login related parameters as these are applied to global state
-        anonymous = kwargs.pop("anonymous", None)
-        force = kwargs.pop("force", None)
-
-        # TODO: move above parameters into apply_init_login
-        settings._apply_init_login(kwargs)
+        init_settings = dict()
+        for key in ["anonymous", "force", "mode", "resume"]:
+            if key in kwargs and kwargs[key] is not None:
+                init_settings[key] = kwargs.get(key)
+        if init_settings:
+            settings.update(init_settings, source=Source.INIT)
 
         if not settings._offline and not settings._noop:
             wandb_login._login(
-                anonymous=anonymous,
-                force=force,
+                anonymous=kwargs.pop("anonymous", None),
+                force=kwargs.pop("force", None),
                 _disable_warning=True,
-                _silent=(settings._quiet or settings._silent) is True,
+                _silent=settings._quiet or settings._silent,
             )
 
         # apply updated global state after login was handled
-        settings._apply_settings(wandb.setup()._settings)
+        settings.apply_settings(wandb.setup().settings)
 
         # get status of code saving before applying user settings
-        save_code_pre_user_settings = settings["save_code"]
+        save_code_pre_user_settings = settings.save_code
 
-        settings._apply_init(kwargs)
+        settings.apply_init(kwargs)
         if not settings._offline and not settings._noop:
             user_settings = self._wl._load_user_settings()
-            settings._apply_user(user_settings)
+            settings.apply_user(user_settings)
 
         # ensure that user settings don't set saving to true
         # if user explicitly set these to false
         if save_code_pre_user_settings is False:
-            settings.update({"save_code": False})
+            settings.update({"save_code": False}, source=Source.INIT)
 
         # TODO(jhr): should this be moved? probably.
-        d = dict(_start_time=time.time(), _start_datetime=datetime.datetime.now(),)
-        settings.update(d)
+        time_stamp: float = time.time()
+        settings.update(
+            {
+                "_start_time": time_stamp,
+                "_start_datetime": datetime.datetime.fromtimestamp(time_stamp),
+            },
+            source=Source.INIT,
+        )
 
         if not settings._noop:
             self._log_setup(settings)
 
             if settings._jupyter:
                 self._jupyter_setup(settings)
 
-        self.settings = settings.freeze()
+        self.settings = settings
+        # self.settings.freeze()
```

Comment:
commented out code

Reply:
ah, this is one of outstanding questions I have and I am hoping to get input from @raubitsj on it. Why bother with freezing at all if everything is now supposed to be consistent, regardless of the order in which individual settings from different sources are loaded?